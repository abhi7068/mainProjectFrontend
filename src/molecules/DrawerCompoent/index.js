import React from 'react';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import CollapseItems from '../../atoms/CollapseItems';
import PropTypes from 'prop-types';
import {
  getFilteredPrice,
  getFilteredDiscount,
  getFilteredRating,
  getResetFilter,
} from '../../store/actions/productAction';
import { connect } from 'react-redux';
const marksforPrice = [
  {
    value: 0,
    label: '0',
  },
  {
    value: 3750,
    label: '3750',
  },
  {
    value: 7500,
    label: '7500',
  },
  {
    value: 15000,
    label: '15000',
  },
];
const marksforRating = [
  {
    value: 0,
    label: '0',
  },
  {
    value: 1,
    label: '1',
  },
  {
    value: 2,
    label: '2',
  },
  {
    value: 3,
    label: '3',
  },
  {
    value: 4,
    label: '4',
  },
  {
    value: 5,
    label: '5',
  },
];
const marksforDiscount = [
  {
    value: 0,
    label: '0',
  },
  {
    value: 20,
    label: '20',
  },
  {
    value: 60,
    label: '60',
  },
  {
    value: 80,
    label: '80',
  },
];
const Compo = (props) => {
  const [outcategories, setoutcategories] = React.useState(false);
  const handleFilterClickOfCategories = () => {
    setoutcategories(!outcategories);
  };
  React.useEffect(() => {
    // if (props.state.product.filterBy === 'FILTER_PRICE') {
    //   setValuePrice(props.state.product.range);
    // } else if (props.state.product.filterBy === 'FILTER_DISCOUNT') {
    //   setValueDiscount(props.state.product.range);
    // } else if (props.state.product.filterBy === 'FILTER_RATING') {
    //   setValueRating(props.state.product.range);
    // // // // } else {
    setValuePrice(props.range.FILTER_PRICE);
    setValueDiscount(props.range.FILTER_DISCOUNT);
    setValueRating(props.range.FILTER_RATING);
    // }
  }, []);
  const [valuePrice, setValuePrice] = React.useState([0, 0]);
  const [valueDiscount, setValueDiscount] = React.useState([0, 0]);
  const [valueRating, setValueRating] = React.useState([0, 0]);
  const handleChangeForPrice = (event, newValue) => {
    setValuePrice(newValue);
    props.getFilteredPrice(newValue);
  };
  const resetFilter = () => {
    setValuePrice([0, 15000]);
    setValueDiscount([0, 80]);
    setValueRating([0, 5]);
    props.getResetFilter();
  };
  const handleChangeForDiscount = (event, newValue) => {
    setValueDiscount(newValue);
    props.getFilteredDiscount(newValue);
  };

  const handleChangeForRating = (event, newValue) => {
    setValueRating(newValue);
    props.getFilteredRating(newValue);
  };
  return (
    <div>
      <ListItem
        data-testid="ListItem"
        button
        onClick={handleFilterClickOfCategories}
        style={{ color: 'black', background: '#969397' }}
      >
        <ListItemIcon style={{ color: 'Black' }}></ListItemIcon>
        <ListItemText
          primary="Filter By Price"
          style={{ color: 'white' }}
          data-testid="test-input"
        />
      </ListItem>
      <CollapseItems
        onchange={handleChangeForPrice}
        min={0}
        max={15000}
        value={valuePrice}
        marks={marksforPrice}
      />
      <ListItem
        button
        onClick={handleFilterClickOfCategories}
        style={{ color: 'black', background: '#969397' }}
      >
        <ListItemIcon style={{ color: 'Black' }}></ListItemIcon>
        <ListItemText primary="Filter By Discount" style={{ color: 'white' }} />
      </ListItem>
      <CollapseItems
        onchange={handleChangeForDiscount}
        min={1}
        max={80}
        value={valueDiscount}
        marks={marksforDiscount}
      />
      <ListItem
        button
        onClick={handleFilterClickOfCategories}
        style={{ color: 'black', background: '#969397' }}
      >
        <ListItemIcon style={{ color: 'Black' }}></ListItemIcon>
        <ListItemText primary="Filter By Rating" style={{ color: 'white' }} />
      </ListItem>
      <CollapseItems
        onchange={handleChangeForRating}
        min={0}
        max={5}
        value={valueRating}
        marks={marksforRating}
      />
      <ListItem
        button
        onClick={resetFilter}
        style={{ color: 'black', background: '#e40046' }}
      >
        <ListItemIcon style={{ color: 'Black' }}></ListItemIcon>
        <ListItemText
          primary="Remove Filters & Sorting "
          style={{ color: 'white' }}
        />
      </ListItem>
      {/* </Collapse> */}
    </div>
  );
};
Compo.propTypes = {
  primary: PropTypes.string,
  typesofdiscount: PropTypes.array,
  getFilteredPrice: PropTypes.func,
  getFilteredDiscount: PropTypes.func,
  getFilteredRating: PropTypes.func,
  state: PropTypes.array,
  getResetFilter: PropTypes.func,
  range: PropTypes.object,
};
const mapStateToProps = (state) => {
  // console.log('');
  return {
    state: state,
    categories: state.categories,
    range: state.product.range,
  };
};
export default connect(mapStateToProps, {
  getFilteredPrice,
  getFilteredDiscount,
  getFilteredRating,
  getResetFilter,
})(Compo);
