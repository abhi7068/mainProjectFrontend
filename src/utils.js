import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import { createMemoryHistory } from 'history';
import thunk from 'redux-thunk';
import PropTypes from 'prop-types';
import reducer from '../src/store/reducers/rootReducer';
import { Router } from 'react-router-dom';

function render(
  ui,
  {
    initialState,
    store = createStore(reducer, initialState, applyMiddleware(thunk)),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    const history = createMemoryHistory();
    return (
      <Provider store={store}>
        <Router history={history}>{children}</Router>
      </Provider>
    );
  }
  Wrapper.propTypes = {
    children: PropTypes.elementType,
  };
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}
export * from '@testing-library/react';
export { render };
