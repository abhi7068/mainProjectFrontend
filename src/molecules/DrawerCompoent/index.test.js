import React from 'react';
import { render, cleanup, fireEvent } from '../../utils';
// import { renderHook, act } from '@testing-library/react-hooks';
import DrawerCompoent from '../DrawerCompoent';

afterEach(cleanup);
it('renders', () => {
  const { asFragment } = render(<DrawerCompoent />);
  expect(asFragment(<DrawerCompoent />)).toMatchSnapshot();
});
test('should make it true', () => {
  // const { result } = renderHook(() => DrawerCompoent());
  const { getByTestId } = render(<DrawerCompoent />);
  const selectNode = getByTestId('test-input');
  expect(selectNode).not.toBeNull();
  fireEvent.click(getByTestId('ListItem'));
  // expect(result.current.outcategories).toBe(true);
});
