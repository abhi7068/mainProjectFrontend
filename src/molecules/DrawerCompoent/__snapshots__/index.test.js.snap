// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div>
    <div
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      data-testid="ListItem"
      role="button"
      style="color: black; background: rgb(150, 147, 151);"
      tabindex="0"
    >
      <div
        class="MuiListItemIcon-root"
        style="color: Black;"
      />
      <div
        class="MuiListItemText-root"
        data-testid="test-input"
        style="color: white;"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          Filter By Price
        </span>
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
    <div
      class="makeStyles-root-53"
    >
      <div
        class="makeStyles-margin-54"
      />
      <span
        class="MuiSlider-root WithStyles(ForwardRef(Slider))-root-55 MuiSlider-colorPrimary MuiSlider-marked"
      >
        <span
          class="MuiSlider-rail WithStyles(ForwardRef(Slider))-rail-60"
        />
        <span
          class="MuiSlider-track WithStyles(ForwardRef(Slider))-track-59"
          style="left: 0%; width: 100%;"
        />
        <input
          type="hidden"
          value="0,15000"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive"
          style="left: 0%;"
        />
        <span
          aria-hidden="true"
          class="MuiSlider-markLabel MuiSlider-markLabelActive"
          style="left: 0%;"
        >
          0
        </span>
        <span
          class="MuiSlider-mark MuiSlider-markActive"
          style="left: 25%;"
        />
        <span
          aria-hidden="true"
          class="MuiSlider-markLabel MuiSlider-markLabelActive"
          style="left: 25%;"
        >
          3750
        </span>
        <span
          class="MuiSlider-mark MuiSlider-markActive"
          style="left: 50%;"
        />
        <span
          aria-hidden="true"
          class="MuiSlider-markLabel MuiSlider-markLabelActive"
          style="left: 50%;"
        >
          7500
        </span>
        <span
          class="MuiSlider-mark MuiSlider-markActive"
          style="left: 100%;"
        />
        <span
          aria-hidden="true"
          class="MuiSlider-markLabel MuiSlider-markLabelActive"
          style="left: 100%;"
        >
          15000
        </span>
        <span
          aria-labelledby="track-inverted-slider"
          aria-orientation="horizontal"
          aria-valuemax="15000"
          aria-valuemin="0"
          aria-valuenow="0"
          aria-valuetext="0째C"
          class="MuiSlider-thumb WithStyles(ForwardRef(Slider))-thumb-56 MuiSlider-thumbColorPrimary PrivateValueLabel-thumb-81"
          data-index="0"
          role="slider"
          style="left: 0%;"
          tabindex="0"
        >
          <span
            class="PrivateValueLabel-offset-83 MuiSlider-valueLabel WithStyles(ForwardRef(Slider))-valueLabel-58"
          >
            <span
              class="PrivateValueLabel-circle-84"
            >
              <span
                class="PrivateValueLabel-label-85"
              >
                0
              </span>
            </span>
          </span>
        </span>
        <span
          aria-labelledby="track-inverted-slider"
          aria-orientation="horizontal"
          aria-valuemax="15000"
          aria-valuemin="0"
          aria-valuenow="15000"
          aria-valuetext="15000째C"
          class="MuiSlider-thumb WithStyles(ForwardRef(Slider))-thumb-56 MuiSlider-thumbColorPrimary PrivateValueLabel-thumb-81"
          data-index="1"
          role="slider"
          style="left: 100%;"
          tabindex="0"
        >
          <span
            class="PrivateValueLabel-offset-83 MuiSlider-valueLabel WithStyles(ForwardRef(Slider))-valueLabel-58"
          >
            <span
              class="PrivateValueLabel-circle-84"
            >
              <span
                class="PrivateValueLabel-label-85"
              >
                15000
              </span>
            </span>
          </span>
        </span>
      </span>
      <div
        class="makeStyles-margin-54"
      />
    </div>
    <div
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      role="button"
      style="color: black; background: rgb(150, 147, 151);"
      tabindex="0"
    >
      <div
        class="MuiListItemIcon-root"
        style="color: Black;"
      />
      <div
        class="MuiListItemText-root"
        style="color: white;"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          Filter By Discount
        </span>
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
    <div
      class="makeStyles-root-53"
    >
      <div
        class="makeStyles-margin-54"
      />
      <span
        class="MuiSlider-root WithStyles(ForwardRef(Slider))-root-55 MuiSlider-colorPrimary MuiSlider-marked"
      >
        <span
          class="MuiSlider-rail WithStyles(ForwardRef(Slider))-rail-60"
        />
        <span
          class="MuiSlider-track WithStyles(ForwardRef(Slider))-track-59"
          style="left: 0%; width: 100%;"
        />
        <input
          type="hidden"
          value="1,80"
        />
        <span
          class="MuiSlider-mark"
          style="left: -1.2658227848101267%;"
        />
        <span
          aria-hidden="true"
          class="MuiSlider-markLabel"
          style="left: -1.2658227848101267%;"
        >
          0
        </span>
        <span
          class="MuiSlider-mark MuiSlider-markActive"
          style="left: 24.050632911392405%;"
        />
        <span
          aria-hidden="true"
          class="MuiSlider-markLabel MuiSlider-markLabelActive"
          style="left: 24.050632911392405%;"
        >
          20
        </span>
        <span
          class="MuiSlider-mark MuiSlider-markActive"
          style="left: 74.68354430379746%;"
        />
        <span
          aria-hidden="true"
          class="MuiSlider-markLabel MuiSlider-markLabelActive"
          style="left: 74.68354430379746%;"
        >
          60
        </span>
        <span
          class="MuiSlider-mark MuiSlider-markActive"
          style="left: 100%;"
        />
        <span
          aria-hidden="true"
          class="MuiSlider-markLabel MuiSlider-markLabelActive"
          style="left: 100%;"
        >
          80
        </span>
        <span
          aria-labelledby="track-inverted-slider"
          aria-orientation="horizontal"
          aria-valuemax="80"
          aria-valuemin="1"
          aria-valuenow="1"
          aria-valuetext="1째C"
          class="MuiSlider-thumb WithStyles(ForwardRef(Slider))-thumb-56 MuiSlider-thumbColorPrimary PrivateValueLabel-thumb-81"
          data-index="0"
          role="slider"
          style="left: 0%;"
          tabindex="0"
        >
          <span
            class="PrivateValueLabel-offset-83 MuiSlider-valueLabel WithStyles(ForwardRef(Slider))-valueLabel-58"
          >
            <span
              class="PrivateValueLabel-circle-84"
            >
              <span
                class="PrivateValueLabel-label-85"
              >
                1
              </span>
            </span>
          </span>
        </span>
        <span
          aria-labelledby="track-inverted-slider"
          aria-orientation="horizontal"
          aria-valuemax="80"
          aria-valuemin="1"
          aria-valuenow="80"
          aria-valuetext="80째C"
          class="MuiSlider-thumb WithStyles(ForwardRef(Slider))-thumb-56 MuiSlider-thumbColorPrimary PrivateValueLabel-thumb-81"
          data-index="1"
          role="slider"
          style="left: 100%;"
          tabindex="0"
        >
          <span
            class="PrivateValueLabel-offset-83 MuiSlider-valueLabel WithStyles(ForwardRef(Slider))-valueLabel-58"
          >
            <span
              class="PrivateValueLabel-circle-84"
            >
              <span
                class="PrivateValueLabel-label-85"
              >
                80
              </span>
            </span>
          </span>
        </span>
      </span>
      <div
        class="makeStyles-margin-54"
      />
    </div>
    <div
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      role="button"
      style="color: black; background: rgb(150, 147, 151);"
      tabindex="0"
    >
      <div
        class="MuiListItemIcon-root"
        style="color: Black;"
      />
      <div
        class="MuiListItemText-root"
        style="color: white;"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          Filter By Rating
        </span>
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
    <div
      class="makeStyles-root-53"
    >
      <div
        class="makeStyles-margin-54"
      />
      <span
        class="MuiSlider-root WithStyles(ForwardRef(Slider))-root-55 MuiSlider-colorPrimary MuiSlider-marked"
      >
        <span
          class="MuiSlider-rail WithStyles(ForwardRef(Slider))-rail-60"
        />
        <span
          class="MuiSlider-track WithStyles(ForwardRef(Slider))-track-59"
          style="left: 0%; width: 100%;"
        />
        <input
          type="hidden"
          value="0,5"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive"
          style="left: 0%;"
        />
        <span
          aria-hidden="true"
          class="MuiSlider-markLabel MuiSlider-markLabelActive"
          style="left: 0%;"
        >
          0
        </span>
        <span
          class="MuiSlider-mark MuiSlider-markActive"
          style="left: 20%;"
        />
        <span
          aria-hidden="true"
          class="MuiSlider-markLabel MuiSlider-markLabelActive"
          style="left: 20%;"
        >
          1
        </span>
        <span
          class="MuiSlider-mark MuiSlider-markActive"
          style="left: 40%;"
        />
        <span
          aria-hidden="true"
          class="MuiSlider-markLabel MuiSlider-markLabelActive"
          style="left: 40%;"
        >
          2
        </span>
        <span
          class="MuiSlider-mark MuiSlider-markActive"
          style="left: 60%;"
        />
        <span
          aria-hidden="true"
          class="MuiSlider-markLabel MuiSlider-markLabelActive"
          style="left: 60%;"
        >
          3
        </span>
        <span
          class="MuiSlider-mark MuiSlider-markActive"
          style="left: 80%;"
        />
        <span
          aria-hidden="true"
          class="MuiSlider-markLabel MuiSlider-markLabelActive"
          style="left: 80%;"
        >
          4
        </span>
        <span
          class="MuiSlider-mark MuiSlider-markActive"
          style="left: 100%;"
        />
        <span
          aria-hidden="true"
          class="MuiSlider-markLabel MuiSlider-markLabelActive"
          style="left: 100%;"
        >
          5
        </span>
        <span
          aria-labelledby="track-inverted-slider"
          aria-orientation="horizontal"
          aria-valuemax="5"
          aria-valuemin="0"
          aria-valuenow="0"
          aria-valuetext="0째C"
          class="MuiSlider-thumb WithStyles(ForwardRef(Slider))-thumb-56 MuiSlider-thumbColorPrimary PrivateValueLabel-thumb-81"
          data-index="0"
          role="slider"
          style="left: 0%;"
          tabindex="0"
        >
          <span
            class="PrivateValueLabel-offset-83 MuiSlider-valueLabel WithStyles(ForwardRef(Slider))-valueLabel-58"
          >
            <span
              class="PrivateValueLabel-circle-84"
            >
              <span
                class="PrivateValueLabel-label-85"
              >
                0
              </span>
            </span>
          </span>
        </span>
        <span
          aria-labelledby="track-inverted-slider"
          aria-orientation="horizontal"
          aria-valuemax="5"
          aria-valuemin="0"
          aria-valuenow="5"
          aria-valuetext="5째C"
          class="MuiSlider-thumb WithStyles(ForwardRef(Slider))-thumb-56 MuiSlider-thumbColorPrimary PrivateValueLabel-thumb-81"
          data-index="1"
          role="slider"
          style="left: 100%;"
          tabindex="0"
        >
          <span
            class="PrivateValueLabel-offset-83 MuiSlider-valueLabel WithStyles(ForwardRef(Slider))-valueLabel-58"
          >
            <span
              class="PrivateValueLabel-circle-84"
            >
              <span
                class="PrivateValueLabel-label-85"
              >
                5
              </span>
            </span>
          </span>
        </span>
      </span>
      <div
        class="makeStyles-margin-54"
      />
    </div>
    <div
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      role="button"
      style="color: black; background: rgb(228, 0, 70);"
      tabindex="0"
    >
      <div
        class="MuiListItemIcon-root"
        style="color: Black;"
      />
      <div
        class="MuiListItemText-root"
        style="color: white;"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          Remove Filters & Sorting 
        </span>
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
  </div>
</DocumentFragment>
`;
